### DISCUSSION TITLE
N&N

### DISCUSSION THREAD

**Discussion by ewillink** [eclipse-justj/justj#29] [Ideas]
The N&N has always been a bit decoupled from its release and is not getting any better.

Surely the Release page should have an N&N link?

A trivial solution today is to just register an "Archive" named "N&N". This could provide a 2-click renavigation from the side bar.

Is there a way to have a 1-click navigation from the release page?

The N&N should logically be under the same source control as the source code. If justj is configured with e.g a `-N&Npath=doc/N&N` justj could link release `X.Y.Z` to `doc/N&N/X.Y.Z` to access a minimal N&N at `doc/N&N/X.Y.Z/index.html`. (N&N should be published from master rather than the associated commit to allow vintage N&Ns to be added to source control.)

**Comment by merks** [comment]
What do you consider to be "the Release page" and what type of content might you expect?  Is this about the p2.manager and something you would like it to generate on some specific pages or about what you're expecting to see on JustJ own pages?

**Comment by ewillink** [comment]
I would like a release page such as [https://download.eclipse.org/modeling/mdt/ocl/builds/release/6.18.0/index.html](url) to link to (the content of) [https://wiki.eclipse.org/OCL/New_and_Noteworthy/2022-12](url) after relocation to [https://git.eclipse.org/c/ocl/org.eclipse.ocl.git/doc/N&N/6.18.0/index.html](url).

Maybe be a more power documentation capability could be available by inserting the N&N immediately above the `This is a p2 update site for` line.

**Comment by merks** [comment]
Generating a "Related Links" like what's on the main download page: 

![image](https://github.com/user-attachments/assets/aeda3675-e8fe-4e3d-aaa5-30c84c2e34aa)

in this section:

![image](https://github.com/user-attachments/assets/db1b1520-bc2e-42a9-ac19-1b41e6dc6881)

would be pretty easy to do and to specify in the arguments.  The links could/would be on all pages and would be the same links on all pages.  How's that?

Note that all pages are generated each time the p2.manager runs so one could change the links any time one generates.

**Comment by ewillink** [comment]
Anything is better than nothing. A general link on the upper (composite) pages is fine, but on a page that links to a specific release zip, it would be nice if the N&N linked to the specific N&N.

(The bulk of the JustJ page is a powerful advert of what JustJ has achieved and can be useful when some problem needs debugging. However from a user point of view, it would be better if the bulk of the central column was delegated to a 'details' page leaving space for more user-oriented content such as documentation/N&N/...)

Yes, regenerating each time is a powerful feature that could allow all pages to upgrade with documentation/N&N if the layout of the documentation/N&N is well planned.
